src/components/popups-etc/tutorial-warning.tsx

```diff
@@ -0,0 +1,39 @@
+import { Signal } from '@preact/signals'
+import Button from '../design-system/button'
+import styles from './tutorial-warning.module.css'
+
+export interface TutorialWarningModalProps {
+	showingTutorialWarning: Signal<boolean>
+	exitTutorial: () => void
+}
+
+export default function TutorialWarningModal(props: TutorialWarningModalProps) {
+	return (
+		<div class={styles.overlay}>
+			<div class={styles.modal}>
+					<div class={styles.stack}>
+						<h2>Are you sure?</h2>
+						<p>
+							Your code will be saved and will be accessible to you.
+							Only the tutorial tab will be removed and cannot be restored.
+						</p>
+					</div>
+
+					<div class={styles.inputRow}>
+						<Button onClick={() => {
+							props.showingTutorialWarning.value = false
+						}}>
+							Cancel
+						</Button>
+						<Button accent onClick={() => {
+							props.exitTutorial()
+							props.showingTutorialWarning.value = false
+						}}>
+							End tutorial
+						</Button>
+					</div>
```

Comment:
Let's change the copy to:

> **Permanently Close Tutorial?**
> 
> Make sure you're finished with this tutorial or don't want to complete it! You won't be able to re-open the tutorial. None of your code will be touched.
>
> [Cancel] [Exit Tutorial]

<details>
<summary>Why?</summary>

- User attention is scarce and the easy path is just to click through the modal, so it's important for the BIG TEXT (the modal's heading) to convey the most important information — in this case, what the action is
- If a user sees a confirmation dialog, they'll probably expect an explanation of what is being confirmed in the body text. If there's only text about what the action *won't* do, it's confusing and maybe feels like the action *does* affect those things (like the user's code, in your text)
- Changed button text to Title Case for consistency with other buttons, and "Exit" instead of "End" for consistency with what this action is called elsewhere
</details>

And make it left-aligned and decrease the button gap to `10px`:

![](https://doggo.ninja/ppb5pf.png)

Reply:
I'm still iffy about putting "Cancel" to the left of "Exit Tutorial" because it looks terribly ugly but it's probably better this way.